### ISSUE TITLE
Spyder Mac apps (for Python 2 and 3) hang on startup with OSX 10.9.5

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#2171]
_From [andy.som...@gmail.com](https://code.google.com/u/112830236860820112250/) on 2015-02-12T10:42:20Z_

Spyder 2.3.2, Python 2.7 will not start correctly on OSX 10.9 (Mavericks)

The app will start, then CPU usage will peg at 100&#37; for about 5-10 minutes, then the app will finally display the splash screen and start normally. 

I have verified this issue on 3 different Mavericks machines. 

Steps to reproduce: 

1: find a computer with OSX 10.9.5
2: download Spyder 2.3.2 dmg
3: attempt to run Spyder

Please note that this is a critical issue, I teach a mathematical modeling course, and I've instructed my students to use Spyder. I'm experiencing this issue myself, and two of my students are unable to use Spyder because of this issue. 

Please let me know if there is any debugging or additional information I can provide.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2171_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-02-12T10:21:07Z_

I'm very sorry to hear that. Right now I'm creating and testing the app in 10.7, and it works well for me.

I've also tested it a couple of times in Mavericks and Yosemite (from friends) and found it works well there too. So I don't know what could be happening in this case.

To get more info, please open Terminal.app and run this command

    /Applications/Spyder.app/Contents/MacOS/Spyder

to see if it reports something fishy.

Also check at the contents of Console.app for messages about Spyder after you start it.

**Summary:** Spyder Mac app hangs on startup with OSX 10.9.5  \<span class="oldvalue"> (was: OSX Spyder hangs on startup) \</span>  
  **Labels:** MS-v2.3.3 OpSys-OSX

**Comment by spyder-issues-migrator** [comment]
_From [andy.som...@gmail.com](https://code.google.com/u/112830236860820112250/) on 2015-02-12T10:52:09Z_

It works fine for me also on 10.7, I think this is likely a PyQt issue, Qt's OSX support has always been pretty grim. I've not heard of any issues from people with 10.10 either. 

The odd thing is that after 2 or 3 hung launches (waiting 5 min each time), after that it seems to work OK.

So, attached are some log dumps.

**Attachment:** [Spyder-Py2_2015-01-26-120332_Sleipnir.hang Spyder-Py2_2015-02-11-135455_Sleipnir.cpu_resource.spin Spyder-Py2_2015-02-11-135951_Sleipnir.hang Spyder-Py2_2015-02-11-140136_Sleipnir.cpu_resource.spin Spyder-Py2_2015-02-12-100524_Sleipnir.cpu_resource.spin](http://code.google.com/p/spyderlib/issues/detail?id=2171#hc2)

**Comment by spyder-issues-migrator** [comment]
_From [hsa...@gmail.com](https://code.google.com/u/103229502451759870663/) on 2015-02-16T10:54:02Z_

I can confirm this too on at least 3 machines from other users. In one case I observed it myself, it took literally 8 minutes to startup. Running on OS X 10.9.5

**Comment by spyder-issues-migrator** [comment]
_From [andy.som...@gmail.com](https://code.google.com/u/112830236860820112250/) on 2015-02-16T11:13:28Z_

I can confirm that disabling AppNap has NO effect on this issue. 

Also, deleting the .spyder config in the home directory, and a clean boot has no effect either. 

Looking at the crash logs, it seems to be hung very deep in some Python code, and it looks like it might be trying to do some file IO.

Attached is another log, after a clean boot, and deleting the .spyder dir. Here, I waited about a 30 minutes before killing the process. 

This is on i7 MacBookPro with OS 10.9.5

**Attachment:** [spyder.log](http://code.google.com/p/spyderlib/issues/detail?id=2171#hc4)

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-02-16T11:55:28Z_

Thanks for the logs. Unfortunately, I can't see anything important in there, just that it seems to be a threading issue.

What happens when you run

    /Applications/Spyder.app/Contents/MacOS/Spyder

from a terminal? Perhaps that's more helpful.

The only thing that occurs to me is to create an app for each MacOS version (since 10.8). But for that I'd need to own a Mac and create the necessary virtual machines. Problem is I don't have the money to do that :-)

**Comment by spyder-issues-migrator** [comment]
_From [andy.som...@gmail.com](https://code.google.com/u/112830236860820112250/) on 2015-02-16T12:00:33Z_

If I run directly from the console, it just hangs, pegs the CPU at 100% and no console output.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-02-16T12:08:18Z_

Too bad. I'll release 2.3.3 later today or tomorrow, let's hope it fixes this by updating Python and its deps.

**Comment by spyder-issues-migrator** [comment]
_From [andy.som...@gmail.com](https://code.google.com/u/112830236860820112250/) on 2015-02-16T12:36:05Z_

I started trying to track this down, looks like it hangs when it creates a new QApplication. 

There are no problems at all with the packaged python or any dependencies, they start perfectly fine when run by themselves. The problem is in QT. 

If you upgrade to a newer QT, that will likely fix the problem. Qt5 does fix a number of issues.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-02-16T16:01:45Z_

**Labels:** -MS-v2.3.3 MS-v2.3.4

**Comment by ccordoba12** [comment]
Hey @AndySomogyi, I hope you're the same one as in Google Code. Let's continue the discussion here :-)

**Comment by AndySomogyi** [comment]
Hi, yes thats me. 

However, after evaluation of a number of other python applications built with PyQT, we have decided that Python is unfortunately not a viable alternative to MATLAB, and have decided to move back to MATLAB.

**Comment by ccordoba12** [comment]
Too bad to hear it. Would you mind to share what were your reasons to finally decide against Python?

**Comment by AndySomogyi** [comment]
95% of the problem was PyQt. Maybe its OK on Windows, but I'm sorry, its train wreck on OSX. It does such a bad job of emulating the native tool kit, that I had to change it to use another skin like CDE or Windows. Combine that with random crashes, and this issue that Spyder takes 30 minutes to start on OSX. 

As a language, I don't have any major issues with Python, well other that the global interpreter lock and the clumsiness of writing C modules. By that I mean that when you write A C module in MATLAB/Octave, you have access to both the lhs and rhs, so you can directly set values in the lhs expression.

**Comment by ccordoba12** [comment]
We moved to 10.9 as the base to create our apps, so I'm confident this bug will be fixed when we release 2.3.5.

**Comment by markushartmann** [comment]
Hello, 

unfortunately I have the same bug in Version 2.3.5.2 at the moment, is there any log I can provide to find the issue?

**Comment by ccordoba12** [comment]
@markushartmann, I don't know what could be happening to be honest. I tested our app in 10.9.0 and 10.10.2 and it's working just fine.

**Comment by markushartmann** [comment]
as already mentioned by the other users it happens when spyder crashes anytime. After a crash is shows this very long startup issue. Reinstalling helps to fix this issue, but reinstalling after every crash is also a time consuming procedure. I try to find anything reproduceable to crash spyder to trigger the bug.

**Comment by zach-morris** [comment]
I had this same issue.  After a random crash of the app, every time I started Spyder-Py2 it became slower and slower to start.  Eventually it took around 10 minutes.  After showing the app contents, I sorted the files by last modified and saw a file named 'pep8' that was very large (around 1GB).  I deleted it and restarted Spyder-Py2 and it started up normally.  Whatever this file is doing was the problem.

**Comment by TimSchmeier** [comment]
I've also experienced this with Yosemite 10.10.5. 10-20m startup time.

**Comment by ccordoba12** [comment]
@TimSchmeier, please run in a terminal

    rm -R ~/.spyder2

Then uninstall and reinstall the application again. That could solve your problem.

**Comment by R-Wright-1** [comment]
I know this is an old thread now, but I've been having this issue with Spyder almost since I installed it over a year ago, and it's actually made me very reluctant to restart my mac ever, as this would mean restarting Spyder. Sometimes Spyder takes over half an hour to open, sometimes it takes even longer and I lose patience and force quit/restart opening it.

I've just found that @zach-morris's solution has fixed this for me, also! I deleted 'pep8', 'pep8.bak' and 'pylint', which were all modified around the same time (approx. 2.5GB in total) and Spyder opened immediately. Thanks so much! I had assumed that this was an issue with having a university managed machine, but evidently not...
